### DISCUSSION TITLE
Is OpenSSL random pool backed by a file?

### DISCUSSION THREAD

**Discussion by afshinpir** [openssl/openssl#25096] [Q&A]
Hi all,

I wonder if OpenSSL random is backed by a persistent file to make sure randomness is guaranteed in the next system boot ups. Does RANDFILE environment variable has an effect here?

Best Regards

**Comment by paulidale** [comment]
It depends on the version of OpenSSL and the RNG being used.

By default, in currently supported versions, the random data will be added to the entropy pool but it won't be credited as actually providing any entropy.  The upshot of this being that OpenSSL will still attempt to gather entropy from the configured sources.
